### PR TITLE
UI: responsive view

### PR DESCRIPTION
**Pull request by DesmondTo**
### Changes made:
Note: the screenshots below are all the mobile view to show how it behaves when the screen is small. You may compare it with your current code to see how I change them.

> 1. Signin page
> - Change the signin card orientation when the screen is small
> - Make the button smaller when the screen is small

Screenshot 2022-11-01 at 8 34 37 PM

> 2. Signup page
> - Responsively scale the width of the signup card, the smaller the screen, the bigger the card so the textfield and button will not overflow
> - Make the button shrink when screen small
Screenshot 2022-11-01 at 8 35 50 PM

> 3. Home page
> - Responsively scale the button (aka shrink the size and change to vertical arrangement if screen too small)
> - Center the decoration (title, image, etc)

Screenshot 2022-11-01 at 8 10 19 PM

> 4. Update profile page
> - No more overflow of textfield
> - No more overflow of icon and text at the top left and top right corner

Screenshot 2022-11-01 at 8 31 37 PM

5. Room page
- When the screen is small, fit the question component to the width of screen to prevent overflow
- This arrangement is similar to that of leetcode
Screenshot 2022-11-02 at 1 05 06 PM

Screenshot 2022-11-02 at 1 10 46 PM

### CODE REVIEW

File: frontend/src/components/forms/SignupForm.module.css [hunk 1]
```diff
@@ -3,7 +3,14 @@
...
-    padding: 8px 75px !important;
-    font-size: 24px !important;
+    padding: 8px 30px !important;
+    font-size: 16px !important;
     font-weight: lighter !important;
-}
\ No newline at end of file
+}
+
+@media only screen and (min-width: 600px) {
```
Comment (hsiaojietng) [inline frontend/src/components/forms/SignupForm.module.css:11]: May I know why 600px is chosen?

**Review by hsiaojietng — COMMENTED** [review]
LGTM, just 1 question!

File: frontend/src/components/forms/SignupForm.module.css (hunk 1 above)
Comment (DesmondTo) [reply frontend/src/components/forms/SignupForm.module.css:11]: 600px is the sm breakpoint of material ui

Comment (hsiaojietng) [reply frontend/src/components/forms/SignupForm.module.css:11]: Ah I did not know that, thanks alot!

**Review by hsiaojietng — APPROVED** [review]
LGTM

**Pull request merged**